File: platypus-js-web-client/src/main/java/com/eas/ui/events/Event.java
package com.eas.ui.events;

import com.google.gwt.dom.client.NativeEvent;

/**
 *
 */
public class Event<T> {

    private NativeEvent event;

    protected Object source;
    private final T target;

    public Event(Object source, T target) {
        this(source, target, null);
    }
    
    public Event(Object source, T target, NativeEvent event) {
        super();
        this.source = source;
        this.target = target;
        this.event = event;
    }

    public NativeEvent getEvent() {
        return event;
    }

    public Object getSource() {
        return source;
    }

    public void setSource(Object source) {
        this.source = source;
    }

    /**
     * Gets the target.
     *
     * @return the target
     */
    public T getTarget() {
        return target;
    }
}

File: platypus-js-web-client/src/app/forms/event.js
define(function () {
    function Event(source, target, event) {
        Object.defineProperty(this, 'event', {
            get: function () {
                return event;
            }
        });
        Object.defineProperty(this, 'source', {
            get: function () {
                return source;
            }
        });
        Object.defineProperty(this, 'target', {
            get: function () {
                if (target && target['p-widget'])
                    return target['p-widget'];
                else
                    return target;
            }
        });
    }
    return Event;
});